### PR TITLE
llm-proxy: response error in JSON format

### PR DESCRIPTION
**Pull request by unknwon**
Followup of https://github.com/sourcegraph/sourcegraph/pull/51226#discussion_r1182231135. The client asks for JSON response and we should respect that.

## Test plan

CI

### CODE REVIEW

File: enterprise/cmd/llm-proxy/shared/main.go [hunk 1]
```diff
@@ -51,7 +51,7 @@ func newHandler(config *Config) http.Handler {
 			r, err := http.NewRequest(http.MethodPost, "https://api.anthropic.com/v1/complete", r.Body)
 			if err != nil {
 				w.WriteHeader(http.StatusInternalServerError)
-				w.Write([]byte(fmt.Sprintf("failed to create request: %s", err)))
+				_, _ = w.Write([]byte(fmt.Sprintf(`{"error": "failed to create request: %s"}`, err)))
```
Comment (eseliger) [inline enterprise/cmd/llm-proxy/shared/main.go:54]: err could contain a `"` and break the JSON, should we instead marshal a 

```go
type errResp struct {
  Error string `json:"error"`
}
```
instead?

Comment (unknwon) [reply enterprise/cmd/llm-proxy/shared/main.go:54]: Oh that's true, but I'd use `strings.Quote` instead to keep the process lightweight (not another JSON marshalling error to handle 😂) WDYT?

Comment (eseliger) [reply enterprise/cmd/llm-proxy/shared/main.go:54]: sure, no idea what other pitfalls there could be with JSON but I guess it'll be alright 😬

Comment (unknwon) [reply enterprise/cmd/llm-proxy/shared/main.go:54]: On a second thought... i'll do the usual way

**Review by eseliger — APPROVED** [review]

**Pull request merged**